jenkins_jobs/modules/parameters.py

```diff
@@ -1143,6 +1143,229 @@ def param_separator(registry, xml_parent, data):
 
 
+def __handle_unochoice_script(data, pdef, script_type):
+    main_script_xml = XML.SubElement(pdef, "script")
+    if script_type == "script":
+        secure_script_xml = XML.SubElement(main_script_xml, "secureScript")
+    elif script_type == "fallback-script":
+        secure_script_xml = XML.SubElement(main_script_xml, "secureFallbackScript")
+
+    sub_script_xml = XML.SubElement(secure_script_xml, "script")
+    groovy_sandbox_xml = XML.SubElement(secure_script_xml, "sandbox")
+    script = data.get(script_type, None)
+    if script:
+        sub_script_xml.text = script["groovy"]
+        groovy_sandbox_xml.text = script["use-groovy-sandbox"]
+        if "script-additional-classpath" in script:
+            classpath_xml = XML.SubElement(secure_script_xml, "classpath")
+            for additional_classpath in script["script-additional-classpath"]:
+                entry_xml = XML.SubElement(classpath_xml, "entry")
+                url_xml = XML.SubElement(entry_xml, "url")
+                url_xml.text = additional_classpath
+
+
+def active_choices_param(registry, xml_parent, data):
+    """yaml: active-choices
+    Active Choices Parameter
+
+    Requires the Jenkins :jenkins-wiki:`Active Choices Plug-in
+    <active+choices+plugin>`.
+
+    :arg str name: Name of the parameter.
+    :arg str description: Description of the parameter.
+    :arg list script: Use a Groovy script to define the parameter.
+
+        :Parameter: * **groovy** (`str`) Groovy DSL Script
+                    * **use-groovy-sandbox** (`bool`) To run this
+                        Groovy script in a sandbox with limited abilities
+                    * **script-additional-classpath** (`list`) Additional
+                        classpath entries accessible from the script.
+    :arg list fallback-script: Use a Fallback script. If the script
+        (specified above) fails, the fallback script will be used as a fallback.
+
+        :Parameter: * **groovy** (`str`) Groovy DSL Script
+                    * **use-groovy-sandbox** (`bool`) To run this Groovy
+                        script in a sandbox with limited abilities.
+                    * **script-additional-classpath** (`list`) Additional
+                        classpath entries accessible from the script.
+    :arg bool enable-filters: If enabled a text box will appear next to
+        this element and will permit the user to filter its entries. The
+        list values never get re-evaluated.
+    :arg int filter-starts-at: How many characters a user must enter
+        before the filter is applied.
+    :arg str choice-type: type of the choices. (default 'PT_SINGLE_SELECT')
+
+        :Allowed Values: * **PT_SINGLE_SELECT**
+                    * **PT_MULTI_SELECT**
+                    * **PT_RADIO**
+                    * **PT_CHECKBOX**
+    :arg str project-name: the name of job from which the user can pick runs
+    :arg str full-project-name: the full project name of job from which the
```

Comment:
Are these two keys interchangeable? Should we specify both? I'm confused.

Reply:
```
<projectName>job-name-test</projectName>
<projectFullName>my_jobs/job_folder1/job-name-test</projectFullName>
```

I don't see them on Jenkins UI, but they are there on XML. I'm not sure how I can handle this otherwise.